### ISSUE TITLE
Column names not correctly translating UTF-8

### ISSUE THREAD

**Issue by CurtHagenlocher** [G-Research/ParquetSharp#205]
Create a test file from Python with

```
import pyarrow as pa
import pyarrow.parquet as pq
schema = pa.schema([
    ('Kein', pa.int32()),
    ('Ärger', pa.string())
])
column1 = pa.array([1, 2, 3], type = pa.int32())
column2 = pa.array(['ä', 'ö', 'ü'], type = pa.string())
batch = pa.RecordBatch.from_arrays([column1, column2], names = schema)
table = pa.Table.from_batches([batch])
pq.write_table(table, 'c:/testdata/test.parquet')

```

Then consume it with ParquetSharp

```
using (var reader = new ParquetFileReader("C:/testdata/test.parquet"))
{
    foreach (var node in reader.FileMetaData.Schema.GroupNode.Fields)
    {
        System.Console.WriteLine(node.Name);
    }
}
```

and the second column name is returned as "A,rger" instead of "Ärger". This is because Marshal.PtrToStringAnsi is being used, whereas I believe the column names are actually encoded as UTF-8.

The test file is also attached.

[Test.zip](https://github.com/G-Research/ParquetSharp/files/6711232/Test.zip)

**Comment by GPSnoopy** [comment]
If that's the case, then quite a few API points need to be tested and modified: 
- node names
- column creation
- column names
- node paths

**Comment by adamreeve** [comment]
I'm closing this as it was fixed with #206.
